app_feup/lib/redux/ActionCreators.dart

```diff
@@ -412,7 +413,20 @@ ThunkAction<AppState> addUserBusStop(Completer<Null> action, BusStop stop){
   return(Store<AppState> store){
     List<BusStop> stops = store.state.content['busstops'];
 
-    stops.add(stop);
+    bool added = false;
+    for(BusStop currStop in stops) {
+      if(currStop.stopCode == stop.stopCode) {
+        added = true;
+        int length = stop.buses.length;
+        for(int i = 0; i < length; i++) {
+          if(!currStop.buses.map((b) => b.busCode).contains(stop.buses[i].busCode))
+            currStop.buses.add(stop.buses[i]);
```

Comment:
This ends up being very confusing and unnecessarily innefficient. If instead of the list `stop.buses` you had a **set**, you would not have the problem of checking if it has already been added. Of course, this would mean you would have to separate the configuration of the stops from the trips themselves, as previously discussed (or implement a comparable, which would just make things confusing). For the trips you would only need to map `stopCode->trips` and for the configuration you would map `stopCode->setOfBusCodes`

In short, you would save the outter for-loop (because `stops` would be a map) and the `.contains` statement, Essentially transforming O(n^3) in O(n), while improving readability and without sacrificing performance elsewhere